e2e_mysql_test.go

```diff
@@ -85,18 +115,18 @@ func TestMySQLDriver(t *testing.T) {
 				}
 				t.Log(now)
 			}
-			cleanup, err := mysql.RegisterDriver(tc.driverName)
+			cleanup, err := mysql.RegisterDriver(tc.driverName, tc.opts...)
 			if err != nil {
 				t.Fatalf("failed to register driver: %v", err)
 			}
 			defer cleanup()
 			cfg := gomysql.NewConfig()
 			cfg.CheckConnLiveness = true
-			cfg.User = mysqlUser
-			cfg.Passwd = mysqlPass
+			cfg.User = tc.user
+			cfg.Passwd = tc.password
```

Comment:
why do we need to be changing the user and pass per test case?

Reply:
The test should use the user/password mysqlUser and mysqlPassword for normal connections. The test should use mysqlIAMUser and "password" when the test includes the option WithIAMAuthN().

Reply:
It looks like the MySQL IAM authN test wasn't even doing and IAM AuthN login previously then... as `mysqlIAMUser` was unused from what I can see?